internal/controller/suite_util_test.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	batch "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gh "github.com/google/go-github/v45/github"
+	mmv1alpha1 "github.com/konflux-ci/mintmaker/api/v1alpha1"
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+
+	. "github.com/konflux-ci/mintmaker/pkg/common"
+	"github.com/konflux-ci/mintmaker/pkg/git/github"
+)
+
+const (
+	// timeout is used as a limit until condition become true
+	// Usually used in Eventually statements
+	timeout  = time.Second * 15
+	interval = time.Millisecond * 250
+)
+
+func createNamespace(name string) {
```

Comment:
Could you add some more comments in this file? It would be nice to have a comment for each function explaining the gist of what it does. Perhaps we don't need that many, but some would be nice.

Reply:
Sure, will update later.